### ISSUE TITLE
RDS Option Groups not assigning add option for SQLSERVER_BACKUP_RESTORE

### ISSUE THREAD

**Issue by JaineshDalicha** [terraform-providers/terraform-provider-aws#4153]
Hi Team,

Terraform version 0.11.7 (Windows)

While creating rds option groups facing issue in adding SQLSERVER_BACKUP_RESTORE options to group while over the console it is working fine after running script.

**Comment by bflad** [comment]
Hi @JaineshDalicha 👋  we will need more information before we can help you.

Can you please provide the information from the issue template including:
* Example configuration
* Expected behavior
* Actual behavior (error messages, plan/apply output, etc)

Thanks!

**Comment by JaineshDalicha** [comment]
Hi Bflad,

Example configuration:
##### IAM ROLE#####

resource "aws_iam_role" "xps-test-rds-role" {
  name = "xps-test-rds-role"

  assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "rds.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
EOF
}
resource "aws_iam_policy" "xps-test-rds-policy" {
    name        = "xps-test-rds-policy"
    description = "RDS S3 policy"
    policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:*"
      ],
      "Effect": "Allow",
      "Resource": "*"
    }
  ]
}
EOF
}
resource "aws_iam_role_policy_attachment" "xps-test-rds-attach" {
    role       = "${aws_iam_role.xps-test-rds-role.name}"
    policy_arn = "${aws_iam_policy.xps-test-rds-policy.arn}"
}
resource "aws_db_option_group" "xps-test-rds-mssql" {
  name                     = "xps-test-rds-mssql"
  option_group_description = "MSSQL Option Group"
  engine_name              = "sqlserver-ex"
  major_engine_version     = "13.00"

  option {
    option_name = "SQLSERVER_BACKUP_RESTORE"

    option_settings {
		name  = "IAM_ROLE_ARN"
		value = "${aws_iam_role.xps-test-rds-role.arn}"
		}
	}
	option {
		option_name = "TDE"
	}
}
################ ERROR #################

error:
Error: Error applying plan:

1 error(s) occurred:

* aws_db_option_group.xps-test-rds-mssql: 1 error(s) occurred:

* aws_db_option_group.xps-test-rds-mssql: Error modifying DB Option Group: InvalidParameterValue: IAM role ARN value is invalid or does not include the required permissions for: SQLSERVER_BACKUP_RESTORE
        status code: 400, request id: 25ee6158-3ad8-41ec-ab41-8698f9f42f99

################# Expected result  ####################
![image](https://user-images.githubusercontent.com/33112650/38609578-1031846c-3d9c-11e8-9162-7fa7d7f69878.png)

**Comment by macbutch** [comment]
Hi @bflad, @JaineshDalicha,

I believe I am seeing something similar here. I'm on the same terraform version but running from MacOS. I see intermittent failures to create the option group (InvalidParameterValue: IAM role ARN value is invalid or does not include the required permissions for: SQLSERVER_BACKUP_RESTORE) but retrying is often successful.

Sometimes the option group is created but the setting is not applied.

Tainting the option group does not work (it can't be deleted because it in use) but Terraform doesn't detect this (it seems to go in a loop and eventually times out unless you modify the option group back to the default in the meantime), then re-run terraform again to attach the new option group to the RDS.

It would be really nice to have something more predictable and less manual here. Is there anything I can provide to help?

**Comment by macbutch** [comment]
Well I sure managed to make a mess of things with what I described above. Now my RDS has two option groups attached (the one that enables native backup and restore and the default one); the native backup restore option does not appear to be effective and running terraform does not remove the default option group (like I think it should).

**Comment by im-dofoborh** [comment]
Guys have a look at this . #763

**Comment by antgel** [comment]
@JaineshDalicha Feel free to edit https://github.com/terraform-providers/terraform-provider-aws/issues/4153#issuecomment-380395103, it's impossible to read.

**Comment by bflad** [comment]
Let's consolidate the discussion and efforts to the earlier reported issue: #763 -- thanks @im-dofoborh! It looks like there's a workaround for the IAM eventual consistency issue described there.
